wfe/web-front-end.go

```diff
@@ -169,27 +183,29 @@ func (wfe *WebFrontEndImpl) HandleFunc(mux *http.ServeMux, pattern string, h fun
 		if err == nil {
 			response.Header().Set("Replay-Nonce", nonce)
 		}
-		response.Header().Set("Access-Control-Allow-Origin", "*")
 
-		// Return a bodyless response to HEAD for any resource that allows GET.
-		if _, ok := methodsOK["GET"]; ok && request.Method == "HEAD" {
-			// We'll be sending an error anyway, but we
-			// should still comply with HTTP spec by not
+		switch request.Method {
+		case "HEAD":
```

Comment:
If handlers are expected to do exactly the same thing for GET and HEAD, that sounds reasonable. It does seem worth logging the real "HEAD" method from `populateRequestEvent`, though. Could move all the `populateRequestEvent` boilerplate into `HandleFunc` and change `request.Method` to GET after that.

Either way, this PR is already dragging on a bit -- wdyt of moving further GET/HEAD stuff to its own issue rather than dealing with it here?